### DISCUSSION TITLE
Bowflex C6 not reporting power nor cadence

### DISCUSSION THREAD

**Discussion by obaceski** [doudar/SmartSpin2k#425] [General]
Hi,
New user here. I was able to connect my heartbeat sensor to SmartSpin2k without any issues
Not much luck with the bike itself. I followed the instructions at the bottom of this page:
https://github.com/doudar/SmartSpin2k/wiki/Pairing-Bluetooth-Sensors
 SmartSpin seems to see the Bike, because when the bike console is off, i only get HR measurements:

[246171][E](BLE_Common): 00 55 <- 0x180d | 0x2a37 | HRS(HRM)[ HR(85) ]
[247572][E](BLE_Common): 00 55 <- 0x180d | 0x2a37 | HRS(HRM)[ HR(85) ]
[248273][E](BLE_Common): 00 55 <- 0x180d | 0x2a37 | HRS(HRM)[ HR(85) ]
[249674][E](BLE_Common): 00 55 <- 0x180d | 0x2a37 | HRS(HRM)[ HR(85) ]

with the bike console on, it starts to report power and cadence, but they are always zero.
Looks like the application is not readint he power and cadence from the bike, for some reason.

Any ideas?

Thanks!

**Comment by doudar** [comment]
What bike is it? 

From my end, if you are able to show a few logging lines of the BLE data coming off of your bike (while you're pedaling) that will help.

Have a great day!

**Comment by obaceski** [comment]
i did look more in detail to the logs, it seems to me the SS2k tryies to connect to the bike and look for the power information, but fails for some reason:
-the address of the bike is fc:30:39:38:07:50

-periodically the SS2k sees the bikes bluuetooth announcements:
[423828][E](BLE_Client): Bluetooth Client Found Devices: {"device 3":{"address":"fc:30:39:38:07:50","name":"IC Bike","UUID":"0x180a"}}

then this happens:
[515528][E](BLE_Client): Bluetooth Client Found Devices: {"device 3":{"address":"fc:30:39:38:07:50","name":"IC Bike","UUID":"0x180a"}}
[516530][E](BLE_Client): Initiating Server Connection
[516531][E](BLE_Client): trying to connect to PM
[516532][E](BLE_Client): Forming a connection to:  fc:30:39:38:07:50
[516532][E](BLE_Client): Reusing Client
[517465][E](BLE_Client): Reconnected client
[517466][E](BLE_Client): Connected to: fc:30:39:38:07:50 RSSI -76
[517539][E](BLE_Client): Failed to find service: 0x1818
[517539][E](BLE_Client): Device Connected
[517540][E](BLE_Client): We are now connected to the BLE Server.
[523542][E](BLE_Client): Initiating Server Connection
[523543][E](BLE_Client): trying to connect to PM
[523543][E](BLE_Client): Forming a connection to:  fc:30:39:38:07:50
[523544][E](BLE_Client): Reusing Client
[523547][E](BLE_Client): Reconnect failed 
[523547][E](BLE_Client): -2 left.


so, it seems the bike announces the "UUID":"0x180a", but for some reason the SS2k does this:
[517466][E](BLE_Client): Connected to: fc:30:39:38:07:50 RSSI -76
[517539][E](BLE_Client): Failed to find service: 0x1818

could it be the SS2k is trying to find service 0x1818 when it should be trying service 0x180a, as announced by the bike?

**Reply by obaceski** [reply]
by the way, forgot to mention, but other apps like zwift can connect to the bike and get a power reading (albeit the inflated Bowflex C6 numbers...) so the bike seems to be working as expected in terms of power readings

**Reply by eMadman** [reply]
Did you build your own unit?  Just curious if you have another ESP32 to test with as well.  

In the meantime, I'll ask my friend to see if he can test with his since I sold my C6 a while back.

**Reply by doudar** [reply]
0x180a is just the general BLE information service. It basically lists what type of device it is. What's interesting is that it appears the bike might be advertising the Cycling Power Service (0x1818) but it doesn't exist on the Schwinns or Bowflex. That's the only way SS2K would attemp to connect to that service. 

Here's a new build from FTMS_Resistance branch which attempts to try to connect the FTMS service first. Please give it a try and report back

Unzip it, make sure it's named exactly firmware.bin and upload it using the webpage. user and pass is admin & admin. It's done when the progress bar reaches 100% 

[firmware.zip](https://github.com/doudar/SmartSpin2k/files/10314840/firmware.zip)

**Reply by obaceski** [reply]
i built my own, in fact is just the ESP32 alone for now (no stepper nor anything else connected to it, just the ESP32 alone)
i just flashed a second ESP32 and i'm getting the same exact result:
-SS2k connects to the heartbeat band without any issues
-it connects but can't find the power meter service on the bike so it disconnects
-SS2k is reachable by Zwift and correctly reports heartbeat but no power (as it didn't find the power meter service)

-if I connect zwift directly to the bike it reads power and cadence fine from it

so it seems that SS2k has trouble reading from the bike, if my deduction is correct (please see logs above)

**Reply by obaceski** [reply]
> 0x180a is just the general BLE information service. It basically lists what type of device it is. What's interesting is that it appears the bike might be advertising the Cycling Power Service (0x1818) but it doesn't exist on the Schwinns or Bowflex. That's the only way SS2K would attemp to connect to that service.
> 
> Here's a new build from FTMS_Resistance branch which attempts to try to connect the FTMS service first. Please give it a try and report back
> 
> Unzip it, make sure it's named exactly firmware.bin and upload it using the webpage. user and pass is admin & admin. It's done when the progress bar reaches 100%
> 
> [firmware.zip](https://github.com/doudar/SmartSpin2k/files/10314840/firmware.zip)

that new firmware worked! 
Thanks a lot!
now SS2k reports heartbeat, cadence and power as expected!

**Reply by doudar** [reply]
Great to know! Thanks for the feedback!
